### DISCUSSION TITLE
Enhancement - Provide options for priority in searching

### DISCUSSION THREAD

**Discussion by gstalnaker** [LMS-Community/slimserver#1177] [General]
Now, by default, when I search my 120,000 track library, the search uses "contains" for its search method which is, if I'm understanding it, is OR'd search terms. With a library this large that is not optimal. I can go to Advanced Search and with a field's dropdown select "Begins with" which works as expected, for example, with Title. It would be advantageous if I could tell the UI that I'd like searches to use a specific option like 'Begins With" or "Contains." rather than having to go into Advanced Search every time to select such an option.

**Comment by michaelherger** [comment]
Settings/Behaviour/Search within words?

**Reply by gstalnaker** [reply]
Michael - I don't think so. Settings/Behavior/Search Within Words allows search terms to find, e.g., "no" in both "nothing" and "volcano." Disabling "Search Within Words" prevents finding volcano in this example. But if I'm searching for "Nothing Can Compare", what I get are search results that include all of the words nothing, can, and compare, a logical OR, e.g., "Nothing OR can OR compare" where what I want is 'nothing AND can AND compare". 

Advanced search provides something close, when I can specify "Starts With" and then "Nothing Can Compare" will find the precise track with such a title. But I have to do multiple clicks in the web UI to enable that option for every search.

My enhancement request was a Search UI option where I can say "use OR" or "use AND" for search terms, or "Set the default to 'Starts With'". Or, perhaps, a search field that understand logical terms like OR and AND?

Since I have your attention (!) - what happens if I enclose search terms in quotations? Is that the equivalent of ANDing the search terms or indicating exact match?
